rasa/engine/storage/storage.py

```diff
@@ -154,7 +154,7 @@ def __post_init__(self) -> None:
     def as_dict(self) -> Dict[Text, Any]:
         """Returns serializable version of the `ModelMetadata`."""
         return {
-            "domain": self.domain.as_dict(),
+            "domain": self.domain.data,
```

Comment:
I think we should keep `domain.data` is a private instance variable (i.e. `._data`) and still access it through `as_dict()`